### PR TITLE
WIP: SB16 POSIX socket test cases update

### PR DESCRIPTION
**Pull request by maxz-lab**
Added more POSIX Socket tests cover more IPv4 and IPv6 cases. Convered host byte to network byte order for port numbers. Added wrapper layer in starboard/shared/moduler for sockaddr definition differences on certain platform.

b/330771976

### CODE REVIEW

File: starboard/nplb/posix_compliance/posix_socket_send_test.cc [hunk 1]
```diff
@@ -77,10 +77,14 @@ TEST(PosixSocketSendTest, RainyDayUnconnectedSocket) {
   EXPECT_TRUE(close(socket_fd) == 0);
 }
 
+// Playstation will terminate the process, so it won't be executed on PS5.
```
[the remark this answers is not in the document]
Comment (maxz-lab) [reply starboard/nplb/posix_compliance/posix_socket_send_test.cc:80]: I will remove it from here, and post it to the test filter file where this specific test gets filtered for PS5.

File: starboard/shared/modular/cobalt_layer_posix_socket_abi_wrappers.cc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+int accept(int sockfd, struct sockaddr* addr, socklen_t* addrlen_ptr) {
+  return __abi_wrap_socket_accept(sockfd, addr, addrlen_ptr);
+}
+
+int __abi_wrap_socket_bind(int sockfd,
+                           const struct sockaddr* addr,
+                           socklen_t addrlen);
+int bind(int sockfd, const struct sockaddr* addr, socklen_t addrlen) {
+  if (addr == NULL) {
```
[the remark this answers is not in the document]
Comment (maxz-lab) [reply starboard/shared/modular/cobalt_layer_posix_socket_abi_wrappers.cc]: I will remove the check in this cobalt_layer_...cc, and access the memory only when the pointer is valid. i.e. let the actual posix function to handle the NULL pointer(s).

File: starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+                                     const char* service,
+                                     const struct addrinfo* hints,
+                                     struct addrinfo** res) {
+  int result = getaddrinfo(node, service, hints, res);
+
+#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
+  struct addrinfo* ai = *res;
+  while (ai != nullptr) {
+    if (ai->ai_addr != nullptr) {
+      // convert the PS5 native sockaddr structure to POSIX sockaddr format
```
Comment (y4vor) [inline starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc:94]: Let's remove references to 1P plarforms

File: starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
+#include <net.h>
+#endif
+
+SB_EXPORT int __abi_wrap_accept(int sockfd,
+                                musl_sockaddr* addr,
+                                socklen_t* addrlen_ptr) {
+#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
+  if (addr != nullptr) {
+    // map musl sockaddr to PS5 native sockaddr, the later has member
```
Comment (y4vor) [inline starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc:27]: same

File: starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+  }
+#endif
+  return accept(sockfd, reinterpret_cast<struct sockaddr*>(addr), addrlen_ptr);
+}
+
+SB_EXPORT int __abi_wrap_bind(int sockfd,
+                              const musl_sockaddr* addr,
+                              socklen_t addrlen) {
+#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
+  // map musl sockaddr to PS5 native sockaddr, the later has member
```
Comment (y4vor) [inline starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc:43]: same

File: starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+#else
+  return bind(sockfd, reinterpret_cast<const struct sockaddr*>(addr), addrlen);
+#endif
+}
+
+SB_EXPORT int __abi_wrap_connect(int sockfd,
+                                 const musl_sockaddr* addr,
+                                 socklen_t addrlen) {
+#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
+  // map musl sockaddr to PS5 native sockaddr, the later has member
```
Comment (y4vor) [inline starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc:65]: same

File: starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+}
+
+SB_EXPORT int __abi_wrap_getifaddrs(struct ifaddrs** ifap) {
+  int result = getifaddrs(ifap);
+#if SB_HAS_QUIRK(SOCKADDR_WITH_LENGTH)
+  struct ifaddrs* ptr = *ifap;
+  struct ifaddrs* last_ptr = ptr;
+  while (ptr != nullptr) {
+    if (ptr->ifa_addr != nullptr) {
+      // convert the PS5 native sockaddr structure to POSIX sockaddr format
```
Comment (y4vor) [inline starboard/shared/modular/starboard_layer_posix_socket_abi_wrappers.cc:115]: same

File: starboard/nplb/posix_compliance/posix_socket_sendto_test.cc [hunk 8]
```diff
@@ -166,13 +166,14 @@ TEST(PosixSocketSendtoTest, RainyDaySendToSocketUntilBlocking) {
 // connected socket to fail to write. For sockets without socket connection
 // support this will show up as a generic error. Otherwise this will show
 // up as a connection reset error.
+// Playstation will terminate the process, so it won't be executed on PS5.
```
Comment (y4vor) [inline starboard/nplb/posix_compliance/posix_socket_sendto_test.cc:169]: Let's remove references to 1P plaltforms.

**Review by y4vor — APPROVED** [review]

**Pull request merged**